source/service.go

```diff
@@ -173,6 +173,19 @@ func (sc *serviceSource) extractHeadlessEndpoints(svc *v1.Service, hostname stri
 
 	targetsByHeadlessDomain := make(map[string][]string)
 	for _, v := range pods.Items {
+		if getReadyPodsOnlyFromAnnotations(svc.Annotations) {
```

Comment:
I don't know if this makes sense to check the annotation from a service for each pod again and again.
I think this needs to be solved in a better way, WDYT?

Reply:
Good point, I will pull it out of the loop.